### PR TITLE
improved error logging for twin failures

### PR DESCRIPTION
**Pull request by YoDaMa**
This PR Solves two issues. 

1) For #587 , this fixes the issue where in MQTT an erroneous Twin value can be set and an error will not be returned. That is the code in the mqtt twin client being added.

2) While reproing for the issue, I noticed that the AMQP error code for an erroneous Twin value was not very verbose. It would just print a generic error:
```
Error: could not update twin reported properties is not retriable
```
With this new code the Error logging is updated to print a full Error status including the message received from the server: 
```
Error: could not update twin reported properties: Provided json is not a valid 'tags' or 'properties' object: arrays are not supported
```

### CODE REVIEW

**Comment by YoDaMa** [comment]
Note: I need to update some of the requirements docs before this PR goes in.

File: device/transport/mqtt/src/mqtt_twin_client.ts [hunk 1]
```diff
@@ -254,9 +254,14 @@ export class MqttTwinClient extends EventEmitter {
...
         const responseBody = message.toString();
+        const parsedMessage = responseBody ? JSON.parse(responseBody) : undefined;
         /*Codes_SRS_NODE_DEVICE_MQTT_TWIN_CLIENT_16_007: [When a message is received on the response topic with an `$rid` property in the query string of the topic matching the one that was sent on the request topic, the `callback` shall be called with a `null` error object and the parsed content of the response message.]*/
         /*Codes_SRS_NODE_DEVICE_MQTT_TWIN_CLIENT_16_017: [When a message is received on the response topic with an `$rid` property in the query string of the topic matching the one that was sent on the request topic, the `callback` shall be called with a `null` error object.]*/
-        requestCallback(null, responseBody ? JSON.parse(responseBody) : undefined);
+        if (parsedMessage && parsedMessage.errorCode) {
+          requestCallback(new Error(parsedMessage.message));
+        } else {
+          requestCallback(null, parsedMessage);
+        }
```
Comment (anthonyvercolano) [inline device/transport/mqtt/src/mqtt_twin_client.ts]: Will be interesting to see how you test....

Comment (YoDaMa) [reply device/transport/mqtt/src/mqtt_twin_client.ts]: thank you for the feedback.

File: device/transport/mqtt/test/_mqtt_twin_client_test.js [hunk 2]
```diff
@@ -266,6 +266,32 @@ describe('MqttTwinClient', function () {
...
+      });
+
+      var twinClient = new MqttTwinClient(fakeMqttBase);
+      var fakeTwin = {
+        fake: 'patch',
+        errorCode: 'fakeErrorCode'
+      };
+      twinClient.updateTwinReportedProperties(fakeTwin, function (err, twin) {
+        assert.instanceOf(err, Error);
+        assert(!twin);
```
Comment (anthonyvercolano) [inline device/transport/mqtt/test/_mqtt_twin_client_test.js]: >assert(!twin); [](start = 8, length = 14)

It has two other assert functions
assert.isNotOk
assert.isOk

Comment (YoDaMa) [reply device/transport/mqtt/test/_mqtt_twin_client_test.js]: noted.

**Pull request merged**